File: app/src/main/java/com/drivingschoolrwandaapp/database/entities/QuestionWithOptions.java
package com.drivingschoolrwandaapp.database.entities;

import androidx.room.Embedded;
import androidx.room.Relation;
import java.util.List;

public class QuestionWithOptions {
    @Embedded
    public TestQuestionEntity question;

    @Relation(
        parentColumn = "id",
        entityColumn = "questionId"
    )
    public List<QuestionOptionEntity> options;
}

File: app/src/main/java/com/drivingschoolrwandaapp/database/entities/QuestionWithOptions.kt
package com.drivingschoolrwandaapp.database.entities

import androidx.room.Embedded
import androidx.room.Relation

data class QuestionWithOptions(
    @JvmField @Embedded var question: TestQuestionEntity? = null,
    @JvmField @Relation(
        parentColumn = "id",
        entityColumn = "questionId"
    )
    var options: List<QuestionOptionEntity>? = null
)
